modules/dev-utils/src/test/java/org/apache/ignite/development/utils/IgniteWalConverterSensitiveDataTest.java

```diff
@@ -0,0 +1,307 @@
+import org.apache.ignite.internal.pagemem.wal.record.WALRecord;
+import org.apache.ignite.internal.processors.cache.CacheObjectImpl;
+import org.apache.ignite.internal.processors.cache.GridCacheOperation;
+import org.apache.ignite.internal.processors.cache.KeyCacheObjectImpl;
+import org.apache.ignite.internal.processors.cache.version.GridCacheVersion;
+import org.apache.ignite.internal.util.typedef.internal.CU;
+import org.apache.ignite.testframework.junits.WithSystemProperty;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.transactions.Transaction;
+import org.junit.Test;
+
+import static java.lang.String.valueOf;
+import static java.lang.System.setOut;
+import static java.lang.System.setProperty;
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_TO_STRING_INCLUDE_SENSITIVE;
+import static org.apache.ignite.cache.CacheAtomicityMode.TRANSACTIONAL;
+import static org.apache.ignite.development.utils.IgniteWalConverter.PRINT_RECORDS;
+import static org.apache.ignite.development.utils.IgniteWalConverter.SENSITIVE_DATA;
+import static org.apache.ignite.development.utils.ProcessSensitiveDataUtils.md5;
+import static org.apache.ignite.testframework.GridTestUtils.assertContains;
+import static org.apache.ignite.testframework.GridTestUtils.assertNotContains;
+import static org.apache.ignite.testframework.wal.record.RecordUtils.isLogEnabled;
+
+/**
+ * Class for testing sensitive data when reading {@link WALRecord} using
+ * {@link IgniteWalConverter}.
+ */
+public class IgniteWalConverterSensitiveDataTest extends GridCommonAbstractTest {
+    /** Sensitive data. */
+    private static final String SENSITIVE_DATA_VALUE = "must_hide_it";
+
+    /** Path to directory where WAL is stored. */
+    private static String WAL_DIR_PATH;
+
+    /** Page size. */
+    private static int PAGE_SIZE;
+
+    /** System out. */
+    private static PrintStream SYS_OUT;
+
+    /**
+     * Test out - can be injected via {@link #injectTestSystemOut()} instead
+     * of System.out and analyzed in test.
+     */
+    private static ByteArrayOutputStream TEST_OUT;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        SYS_OUT = System.out;
+        TEST_OUT = new ByteArrayOutputStream(16 * 1024);
+
+        int nodeId = 0;
+
+        IgniteEx crd = startGrid(nodeId);
+        crd.cluster().active(true);
+
+        try (Transaction tx = crd.transactions().txStart()) {
+            crd.cache(DEFAULT_CACHE_NAME).put(SENSITIVE_DATA_VALUE, SENSITIVE_DATA_VALUE);
+            tx.commit();
+        }
+
+        GridKernalContext kernalCtx = crd.context();
+        IgniteWriteAheadLogManager wal = kernalCtx.cache().context().wal();
+
+        for (WALRecord walRecord : withSensitiveData()) {
+            if (isLogEnabled(walRecord))
+                wal.log(walRecord);
+        }
+
+        wal.flush(null, true);
```

Comment:
These tests look too synthetic with this manual write to WAL. Could you please rewrite it and add some real cache puts and SQL insert queries that include sensitive data?